genet/core.py

```
@@ -2207,38 +2207,42 @@ def validation_report_for_node_elevation(self, low_limit=-50, mont_blanc_height=
-            elif node_elev > mont_blanc_height:
-                too_high[node_id] = node_elev
-
-        report = {
-            'summary': {'total_nodes': len(elevation_list),
-                        'min_value': int(min_value),
-                        'max_value': int(max_value),
-                        'mean': int(mean),
-                        'median': int(median),
-                        'extremely_high_values_count': len(too_high),
-                        'extremely_low_values_count': len(too_low)},
-
-            'values': {'extremely_high_values_dict': too_high,
-                       'extremely_low_values_dict': too_low}}
+        if 'z' in list(nodes.columns):
+            elevation_list = nodes['z'].to_list()
+            min_value = np.min(elevation_list)
+            max_value = np.max(elevation_list)
+            mean = np.mean(elevation_list)
+            median = np.median(elevation_list)
+
+            nodes_dictionary = dict(zip(nodes['id'], nodes['z']))
+            too_high = {}
+            too_low = {}
+
+            for node_id, node_elev in nodes_dictionary.items():
+                if node_elev < low_limit:
+                    too_low[node_id] = node_elev
+                elif node_elev > mont_blanc_height:
+                    too_high[node_id] = node_elev
+
+            report = {
+                'summary': {'total_nodes': len(elevation_list),
+                            'min_value': int(min_value),
+                            'max_value': int(max_value),
+                            'mean': int(mean),
+                            'median': int(median),
+                            'extremely_high_values_count': len(too_high),
+                            'extremely_low_values_count': len(too_low)},
+
+                'values': {'extremely_high_values_dict': too_high,
+                           'extremely_low_values_dict': too_low}}
+
+            return report
 
-        return report
+        else:
```

I don't like the silent failure approach. We should raise an exception here with a suitable message. As it is, the caller of this code could be assigning `None` to some variable for the report without knowing that had happened until something subsequently blows up. If we raise an exception, the failure is explicit. We should also have a unit test for this case.